### PR TITLE
Added run-rerun command, for executing tests in more than one test suite run

### PR DESCRIPTION
**Pull request by Ilrilan**
## Motivation/Description of the PR

Applicable helpers:

- [ ] Webdriver
- [ ] Puppeteer
- [ ] Nightmare
- [ ] REST
- [ ] Appium
- [ ] Protractor
- [ ] TestCafe

- Description of this PR, which problem it solves
- A link to the corresponding issue (if applicable).

## Type of change

- [ ] Breaking changes
- [x] New functionality
- [ ] Bug fix
- [ ] Documentation changes/updates
- [ ] Hot fix
- [ ] Markdown files fix - not related to source code

## Checklist:

- [x] Tests have been added
- [x] Documentation has been added (Run `npm run docs`)
- [x] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

### CODE REVIEW

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/command/rerun/codecept_rerunner.js  4
- lib/command/run-rerun.js  5
         

Clones added
============
- lib/command/rerun/codecept_rerunner.js  1
- lib/command/run-rerun.js  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4715259&bid=15746965)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/codecept_rerunner.js  4
- lib/command/run-rerun.js  3
         

Complexity decreasing per file
==============================
+ lib/command/run.js  -2
         

Clones added
============
- lib/codecept_rerunner.js  1
- lib/command/utils.js  1
- lib/command/run-rerun.js  3
- lib/command/run.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4715259&bid=15746965)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/codecept_rerunner.js  4
- test/data/sandbox/configs/run-rerun/first_test.js  1
- test/runner/run_rerun_test.js  1
- lib/command/run-rerun.js  3
- test/data/sandbox/configs/run-rerun/customHelper.js  1
         

Complexity decreasing per file
==============================
+ lib/command/run.js  -2
         

Clones added
============
- lib/command/utils.js  1
- lib/codecept_rerunner.js  1
- test/runner/run_rerun_test.js  1
- lib/command/run-rerun.js  3
- lib/command/run.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4715259&bid=15746965)

File: lib/codecept_rerunner.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      } catch (e) {
+        reject(e);
+      }
+    });
+  }
+
+  async runTests(test) {
+    const configRerun = this.config.rerun || {};
+    const minSuccess = configRerun.minSuccess || 1;
+    const maxReruns = configRerun.maxReruns || 1;
```
Comment (Vorobeyko) [inline lib/codecept_rerunner.js]: Need check if minSuccess more then maxReruns. And throw Error

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- test/data/sandbox/configs/run-rerun/first_test.js  1
- test/runner/run_rerun_test.js  2
- lib/command/run-rerun.js  3
- lib/codecept_rerunner.js  4
- test/data/sandbox/configs/run-rerun/first_ftest.js  2
- test/data/sandbox/configs/run-rerun/customHelper.js  1
         

Complexity decreasing per file
==============================
+ lib/command/run.js  -2
         

Clones added
============
- test/data/sandbox/configs/run-rerun/codecept.conf.min_less_max.js  1
- test/runner/run_rerun_test.js  1
- lib/command/run-rerun.js  3
- lib/codecept_rerunner.js  1
- lib/command/utils.js  1
- lib/command/run.js  1
- test/data/sandbox/configs/run-rerun/codecept.conf.fail_test.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4715259&bid=15746965)

**Comment by Vorobeyko** [comment]
> I’m curious how the tests reports look like in the end.

It looks like retries tests. See screenshot:
image

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- test/runner/run_rerun_test.js  2
- lib/command/run-rerun.js  3
- test/data/sandbox/configs/run-rerun/first_test.js  1
- test/data/sandbox/configs/run-rerun/first_ftest.js  2
- test/data/sandbox/configs/run-rerun/customHelper.js  1
- lib/codecept_rerunner.js  4
         

Complexity decreasing per file
==============================
+ lib/command/run.js  -2
         

Clones added
============
- test/runner/run_rerun_test.js  1
- lib/command/run-rerun.js  3
- lib/command/run.js  1
- test/data/sandbox/configs/run-rerun/codecept.conf.min_less_max.js  1
- test/data/sandbox/configs/run-rerun/codecept.conf.fail_test.js  1
- lib/command/utils.js  1
- lib/codecept_rerunner.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4715259&bid=15746965)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- lib/command/run-rerun.js  3
- lib/codecept_rerunner.js  4
- test/runner/run_rerun_test.js  2
- test/data/sandbox/configs/run-rerun/first_test.js  1
- test/data/sandbox/configs/run-rerun/first_ftest.js  2
- test/data/sandbox/configs/run-rerun/customHelper.js  1
         

Complexity decreasing per file
==============================
+ lib/command/run.js  -2
         

Clones added
============
- lib/command/run.js  1
- lib/command/run-rerun.js  3
- lib/codecept_rerunner.js  1
- test/runner/run_rerun_test.js  1
- test/data/sandbox/configs/run-rerun/codecept.conf.fail_test.js  1
- test/data/sandbox/configs/run-rerun/codecept.conf.min_less_max.js  1
- lib/command/utils.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4715259&bid=15746965)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/command/run-rerun.js  3
- lib/codecept_rerunner.js  4
- test/data/sandbox/configs/run-rerun/customHelper.js  1
- test/data/sandbox/configs/run-rerun/first_ftest.js  2
- test/data/sandbox/configs/run-rerun/first_test.js  1
- test/runner/run_rerun_test.js  2
         

Complexity decreasing per file
==============================
+ lib/command/run.js  -2
         

Clones added
============
- lib/command/run-rerun.js  3
- lib/codecept_rerunner.js  1
- lib/command/run.js  1
- test/data/sandbox/configs/run-rerun/codecept.conf.min_less_max.js  1
- lib/command/utils.js  1
- test/data/sandbox/configs/run-rerun/codecept.conf.fail_test.js  1
- test/runner/run_rerun_test.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/DavertMik/CodeceptJS/pullRequest?prid=4715259&bid=15746965)

**Pull request merged**